pkg/devicerepository/grpc.go

```diff
@@ -50,8 +52,8 @@ func (dr *DeviceRepository) ListBrands(ctx context.Context, req *ttnpb.ListEndDe
 	if err := rights.RequireApplication(ctx, req.ApplicationIDs, ttnpb.RIGHT_APPLICATION_DEVICES_READ); err != nil {
 		return nil, err
 	}
-	if req.Limit == 0 {
-		req.Limit = 1000
+	if req.Limit > defaultLimit || req.Limit <= 0 {
```

Comment:
Nit; `Limit` is unsigned

```suggestion
	if req.Limit > defaultLimit || req.Limit == 0 {
```